fastembed/embedding.py

```diff
@@ -295,24 +318,30 @@ def download_file_from_gcs(cls, url: str, output_path: str, show_progress: bool
         return output_path
 
     @classmethod
-    def download_files_from_huggingface(cls, repod_id: str, cache_dir: Optional[str] = None) -> str:
+    def download_files_from_huggingface(cls, repo_ids: List[str], cache_dir: Optional[str] = None) -> str:
         """
         Downloads a model from HuggingFace Hub.
         Args:
-            repod_id (str): The HF hub id (name) of the model to retrieve.
+            repo_id (str): The HF hub id (name) of the model to retrieve.
             cache_dir (Optional[str]): The path to the cache directory.
         Raises:
             ValueError: If the model_name is not in the format <org>/<model> e.g. "jinaai/jina-embeddings-v2-small-en".
         Returns:
             Path: The path to the model directory.
         """
-        from huggingface_hub import snapshot_download
 
-        return snapshot_download(
-            repo_id=repod_id,
-            ignore_patterns=["model.safetensors", "pytorch_model.bin"],
-            cache_dir=cache_dir,
-        )
+        for index, repo_id in enumerate(repo_ids):
+            try:
+                return snapshot_download(
+                    repo_id=repo_id,
+                    ignore_patterns=["model.safetensors", "pytorch_model.bin"],
+                    cache_dir=cache_dir,
+                )
+            except (RepositoryNotFoundError, EnvironmentError) as e:
+                logger.error(f"Failed to download model from HF source: {repo_id}: {e} ")
+                if repo_id == repo_ids[-1]:
+                    raise e
+                logger.info(f"Trying another source: {repo_ids[index+1]}")
 
     @classmethod
     def decompress_to_cache(cls, targz_path: str, cache_dir: str):
```

Comment:
This cache should now download the model to `~/. cache/fastembed/` to be analogous to huggingface hub which downloads to `~/. cache/huggingface/hub`

Reply:
I believe since we specify the default path `~/.cache/fastembed/ ` to HF's `snapshot_download()` method, it is already routed there.

Reply:
Or do you mean, update https://github.com/qdrant/fastembed/blob/3b32619a4c1e3f958bedeae9a437a675aee91d1d/fastembed/embedding.py#L476

to `~/.cache/fastembed/`?

Reply:
Yes, this is the change. Move it from tempdir to the dedicated cache dir and have all the models downloaded there — including those from the hub and URL/GCS both

Reply:
Will do.

Reply:
To note, `tempfile.gettempdir()` returns the system's OS specific cache dir. Location like `~.cache`,  not available on Windows.